File: social-model/src/main/java/com/socialuni/social/model/model/QO/FrontErrorLogVO.java
package com.socialuni.social.model.model.QO;

import com.socialuni.social.entity.model.DO.log.FrontErrorLogDO;
import com.socialuni.social.entity.model.DO.user.UserDO;
import lombok.Data;

import java.util.Date;

/**
 * @author qinkaiyuan
 * @date 2018-11-18 20:48
 */
//记录前端错误日志，不该出现的错误
@Data
public class FrontErrorLogVO {
    //出错的url
    private String uri;

    //出错的业务备注
    private String detail;

    //传入了什么参数导致报错
    private String params;

    //错误信息
    private String errorMsg;
    private String platform;
    private String provider;
    private String appVersion;


}

File: socialuni-js/packages/socialuni-api-base/src/model/FrontErrorLogVO.ts
import PlatformType from "socialuni-constant/constant/PlatformType";
import RouterUtil from "socialuni-util/src/util/RouterUtil";

export default class FrontErrorLogVO {
    uri: string = null
    detail: string = null
    params: string = null
    errorMsg: string = null
    platform: string = null
    provider: string = null
    frontPage: string = null
    appVersion: number = null

    constructor(uri: string, detail: string, params: string, errorMsg: string) {
        this.uri = uri
        this.detail = detail
        this.params = params
        this.errorMsg = errorMsg
        this.frontPage = RouterUtil.getCurrentPageURI()
        this.platform = socialuniSystemModule.platform
        if (this.platform === PlatformType.mp) {
            this.provider = socialuniSystemModule.provider
        } else {
            this.appVersion = socialuniSystemModule.appVersion
        }
    }
}
